bazel/grpc_build_system.bzl

```diff
@@ -305,46 +317,69 @@ def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data
         "flaky": flaky,
         "linkstatic": linkstatic,
     }
-    if uses_polling:
-        # the vanilla version of the test should run on platforms that only
-        # support a single poller
-        native.cc_test(
-            name = name,
-            testonly = True,
-            tags = (tags + [
-                "no_linux",  # linux supports multiple pollers
-            ]),
-            **args
-        )
-
-        # on linux we run the same test multiple times, once for each poller
-        for poller in POLLERS:
-            native.sh_test(
-                name = name + "@poller=" + poller,
-                data = [name] + data,
-                srcs = [
-                    "//test/core/util:run_with_poller_sh",
-                ],
-                size = size,
-                timeout = timeout,
-                args = [
-                    poller,
-                    "$(location %s)" % name,
-                ] + args["args"],
-                tags = (tags + ["no_windows", "no_mac"]),
-                exec_compatible_with = exec_compatible_with,
-                exec_properties = exec_properties,
-                shard_count = shard_count,
-                flaky = flaky,
-            )
-    else:
-        # the test behavior doesn't depend on polling, just generate the test
-        native.cc_test(name = name, tags = tags + ["no_uses_polling"], **args)
     ios_cc_test(
         name = name,
         tags = tags,
         **args
     )
+    if not uses_polling:
+        # the test behavior doesn't depend on polling, just generate the test
+        # TODO(hork): identify if any non-polling tests should be exercised by
+        # all known EventEngines. It is assumed that non-polling tests are
+        # engine-agnostic.
+        native.cc_test(name = name, tags = tags + ["no_uses_polling"], **args)
+        return
+
+    # -- Tests that exercise the polling system --
+
+    # Non-Linux platforms
+    for engine in EVENT_ENGINES:
+        if "no_windows" not in engine["tags"] and "no_mac" not in engine["tags"]:
+            # These platforms do not support multiple polling engines,
+            # so just create a target for each EventEngine.
+            native.cc_test(
+                # TODO(hork): these names are depended upon in cpp end2end
+                # tests data arguments. Those tests need to be updated to pull
+                # the right dependencies.
+                # name = name + "_engine_" + engine["name"],
+                name = name,
+                tags = (tags + engine["tags"] + [
+                    "no_linux",  # linux supports multiple pollers
+                ]),
+                env = {"GRPC_EVENTENGINE_STRATEGY": engine["name"]},
```

Comment:
I did some experimentation around that (see commit history), but it would require replacing all sh_tests with cc_tests, and I decided that it may be best to consider the repercussions of that in separate work. This code can be considered temporary, but it's not terribly harmful even if it isn't.

Reply:
* ... would require replacing all sh_tests with cc_tests in the cpp/end2end tests as well.

Reply:
Now that we're converting from sh_test to cc_test anyway, I think we can go ahead and do this.